core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -125,15 +127,82 @@ public ListenableFuture<Void> execute(
                 bindParameters(statement, parameters),
                 true);
 
-        ColumnMetadata column = ColumnMetadata.builder()
-                .setName(element.getName().getValue())
-                .setType(type)
-                .setNullable(element.isNullable())
-                .setComment(element.getComment())
-                .setProperties(columnProperties)
-                .build();
+        if (element.getField().getParts().size() == 1) {
+            if (columnHandles.containsKey(columnName.toLowerCase(ENGLISH))) {
+                if (!statement.isColumnNotExists()) {
+                    throw semanticException(COLUMN_ALREADY_EXISTS, statement, "Column '%s' already exists", columnName);
+                }
+                return immediateVoidFuture();
+            }
+            ColumnMetadata column = ColumnMetadata.builder()
+                    .setName(columnName)
+                    .setType(type)
+                    .setNullable(element.isNullable())
+                    .setComment(element.getComment())
+                    .setProperties(columnProperties)
+                    .build();
+            plannerContext.getMetadata().addColumn(session, tableHandle, column);
+        }
+        else {
```

Comment:
We have to independent execution paths.
Would it make sense to split `AddColumnTask.java` into `AddColumnTask.java` and `Add[Column]FieldTask.java`?